datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/auth/CreateAccessTokenResolver.java

```diff
@@ -0,0 +1,106 @@
+    return CompletableFuture.supplyAsync(() -> {
+      final QueryContext context = environment.getContext();
+      final CreateAccessTokenInput input = bindArgument(environment.getArgument("input"), CreateAccessTokenInput.class);
+
+      log.info("User {} requesting new access token for user {} ", context.getActorUrn(), input.getActorUrn());
+
+      if (isAuthorizedToGenerateToken(context, input)) {
+        final TokenType type = TokenType.valueOf(input.getType().toString()); // warn: if we are out of sync with AccessTokenType there are problems.
+        final String actorUrn = input.getActorUrn();
+        final Date date = new Date();
+        final long createdAtInMs = date.getTime();
+        final long expiresInMs = AccessTokenUtil.mapDurationToMs(input.getDuration());
+
+        final String tokenName = input.getName() != null ? input.getName() : "";
+        final String tokenDescription = input.getDescription() != null ? input.getDescription() : "";
+
+        final String accessToken = _statefulTokenService.generateAccessToken(type,
+            createActor(input.getType(), actorUrn), expiresInMs, createdAtInMs, tokenName, tokenDescription,
+            context.getActorUrn());
+        log.info("Generated access token for {} of type {} with duration {}", input.getActorUrn(), input.getType(),
+            input.getDuration());
+        try {
+          final String tokenHash = _statefulTokenService.hash(accessToken);
+
+          final AccessToken result = new AccessToken();
+          result.setAccessToken(accessToken);
+          final AccessTokenMetadata metadata = new AccessTokenMetadata();
+          metadata.setUrn(Urn.createFromTuple(Constants.ACCESS_TOKEN_ENTITY_NAME, tokenHash).toString());
+          metadata.setType(EntityType.ACCESS_TOKEN);
+          result.setMetadata(metadata);
+
+          return result;
+        } catch (Exception e) {
+          throw new RuntimeException(String.format("Failed to create new access token with name %s", input.getName()), e);
+        }
+      }
+      throw new AuthorizationException("Unauthorized to perform this action. Please contact your DataHub administrator.");
+    });
+  }
+
+  private boolean isAuthorizedToGenerateToken(final QueryContext context, final CreateAccessTokenInput input) {
+    // Currently only an actor can generate a personal token for themselves.
+    if (AccessTokenType.PERSONAL.equals(input.getType())) {
+      return isAuthorizedToGeneratePersonalAccessToken(context, input);
+    }
+    throw new UnsupportedOperationException(String.format("Unsupported AccessTokenType %s provided", input.getType()));
+  }
+
+  private boolean isAuthorizedToGeneratePersonalAccessToken(final QueryContext context, final CreateAccessTokenInput input) {
+    if (input.getActorUrn().equals(context.getActorUrn())) {
+      return AuthorizationUtils.canGeneratePersonalAccessToken(context) || AuthorizationUtils.canManageAllPersonalTokens(context);
```

Comment:
Can we simplify this conditional at all? Seems it can be better factored, even though technically speaking it's correct